source/dub/packagesuppliers/registry.d

```diff
@@ -86,24 +86,21 @@ class RegistryPackageSupplier : PackageSupplier {
 			m_metadataCache.remove(packageId);
 		}
 
-		auto url = m_registryUrl ~ NativePath("api/packages/" ~ packageId ~ "/info?minimize=true");
+		auto url = m_registryUrl ~ NativePath("api/packages/infos?packages=[\"" ~
+				packageId ~ "\"]&include_dependencies=true&minimize=true");
 
 		logDebug("Downloading metadata for %s", packageId);
 		string jsonData;
 
-		try
-			jsonData = cast(string)retryDownload(url);
-		catch(HTTPStatusException e) {
-			if (e.status == 404) {
-				logDebug("Package %s not found at %s (404): %s", packageId, description, e.msg);
-				return Json(null);
```

Comment:
This was really bad, because at the moment the https://dub-registry.herokuapp.com returned 404 for the package. This lead to a null Json being returned and thus the mirror request being accepted as successful.